draft-packages/select/KaizenDraft/Select/Select.tsx

```diff
@@ -14,7 +14,71 @@ const styles = require("./styles.react.scss")
 
 export { ValueType } from "react-select"
 
-export const Select = (props: ReactSelectProps) => {
+export type SelectProps = {
+  /**
+   * The secondary variant is a more subdued variant that takes up as little space as possible
+   * `variant="secondary" reversed="false" is not implemented and will throw a "not implemented" error
+   * @default "default"
+   */
+  variant?: VariantType
+
+  /**
+   * Use a reversed colour scheme
+   * `variant="default" reversed="true" is not implemented and will throw a "not implemented" error
+   * @default false
+   */
+  reversed?: boolean
+
+  /**
+   * Whether the "control" (the button you click to open the menu) width fills the
+   * container or is as wide as the selected option text. Note that the control text
+   * will ellipsize if it is wider than the parent container.
+   * `variant="default" selectControlWidth="containSelection"` is not implemented and
+   * will throw a "not implemented" error
+   * @default "containSelection" if variant is "secondary", otherwise "fillContainer"
+   */
+  selectControlWidth?: SelectControlWidthType
+}
+
+export type VariantType = "default" | "secondary"
+export type SelectControlWidthType = "fillContainer" | "containSelection"
+
+export const Select = (props: SelectProps & ReactSelectProps) => {
+  if (
+    props.selectControlWidth === "containSelection" &&
+    props.variant !== "secondary"
+  ) {
+    throw new Error(
+      `the prop selectControlWidth=containSelection is not yet implemented when variant="default"`
+    )
+  }
+  const { variant = "default", reversed = false } = props
+
+  // the default for selectControlWidth depends on the variant
+  const selectControlWidth =
+    props.selectControlWidth != null
+      ? props.selectControlWidth
+      : variant === "secondary"
+      ? "containSelection"
+      : "fillContainer"
```

Comment:
I hate this code, but I'm not sure what the better approach is? Prettier strips away brackets I tried to add to make it more readable (because of the gross nested ternaries). The ternary `?`/`:` is the only "expression operator" available in JS, otherwise I'd have to use `let` and `if` statements (which aren't expressions) which feels wrong. Any better ideas?

Reply:
No better way that I'm aware of. Splitting out to separate statements isn't too terrible. I don't have any preference to either approach 🤷‍♀️
```
let secondaryWidth = variant === "secondary"
      ? "containSelection"
      : "fillContainer";

  const selectControlWidth =
    props.selectControlWidth != null
      ? props.selectControlWidth
      : secondaryWidth
```